aks/secure-baseline/GitHubAction/AKS-deploy.yml

```
@@ -0,0 +1,45 @@
+# This is a basic workflow to help you get started with Actions
+
+name: CI
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the master branch
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+# Environment Variables
+env:
+  RGNAMECLUSTER: "{YOUR-CLUSTER-RG-NAME_TO_BE_CREATED}"
+  RGLOCATION: "{REGION}"
+  REDUNDANCY_LOCATION: "{REDUDANCY_REGION}"
+  CLUSTER_VNET_RESOURCE_ID: "/subscriptions/[subscriptionId]/resourceGroups/far4-rg-enterprise-networking-spokes/providers/Microsoft.Network/virtualNetworks/vnet-hub-spoke-BU0001A0008-00"
+  k8sRbacAadProfileTenantId: "{GUID}"
+  k8sRbacAadProfileAdminGroupObjectID: "{GUID}"
+  FIREWALL_SUBNET_RESOURCEID: "[]"
+  
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+
+# Login into your Azure Subscription using your Azure credentials - make sure the credentials has write permissions for the specific resource group/subscription. The credentials should be stored in GitHub Secrets - (Go to Settings Tab ->Secrets)
+# More details on how to create the SPN can be found here - https://github.com/Azure/login#configure-deployment-credentials
+    - name: Azure Login
+      uses: Azure/login@v1.1
+      with:
+        creds: ${{ secrets.AZURE_CREDENTIALS }}
+
+# Deploy the cluster into your environment, assuming all prerequesits are up and running, k8sRbacAadProfileServerAppId, k8sRbacAadProfileServerAppSecret, k8sRbacAadProfileClientAppId, and k8sRbacAadProfileTenantId are stored in the GitHub Secrets (Go to Settings->Secrets)
+    - name: Azure CLI Action
+      uses: Azure/cli@v1.0.0
+      with:
+        # Specify the script here
+        inlineScript: |
+          az group create --name ${{ env.RGNAMECLUSTER }} --location ${{ env.RGLOCATION }}
```

```suggestion
          az group create --name cluster-stamp-bu0001a0008 --location eastus2
```

when writing content about this, I think I will be easier for our Technical Writers to have a story to tell, that's why I suggest this change from parametric resource group to a fixed one which is aligned to other pieces of AKS Bicycle.

---

If I remember correctly, we talked about it that we should be able to redeploy a new cluster in case needed. Using Environment variables makes it more flexible - having one central place to update the variables. Moving forward we should add the option to configure this outside of the Action by using GitHub Inputs - https://help.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#inputs The reason I used environment variables is because GitHub Input stores in environment variables.

---

Please let me mark this as resolved. I will ping you offline for the rest of the suggested changes.